httpx/dispatch/base.py

```diff
@@ -96,16 +45,16 @@ def send(
     ) -> Response:
         raise NotImplementedError()  # pragma: nocover
 
-    def close(self) -> None:
+    async def close(self) -> None:
         pass  # pragma: nocover
 
-    def __enter__(self) -> "Dispatcher":
+    async def __aenter__(self) -> "Dispatcher":
         return self
 
-    def __exit__(
+    async def __aexit__(
         self,
         exc_type: typing.Type[BaseException] = None,
         exc_value: BaseException = None,
         traceback: TracebackType = None,
     ) -> None:
-        self.close()
+        await self.close()
```

Comment:
Interesting diff on this class, right? Related to the codegen approach folks from urllib3-async pointed out :)